backend/init/init.go

```diff
@@ -38,6 +39,7 @@ func init() {
 		"consul": func() backend.Backend { return backendconsul.New() },
 		"inmem":  func() backend.Backend { return backendinmem.New() },
 		"s3":     func() backend.Backend { return backendS3.New() },
+		"azure":  func() backend.Backend { return backendAzure.New() },
```

Comment:
@pmcatominey could we take @jbardin's suggestion of adding another entry for AzureRM and updating the docs? This otherwise LGTM :)